src/sort.rs

```diff
@@ -68,7 +72,26 @@ pub fn sort_toml(
     let mut first_table = None;
     let mut heading_order: BTreeMap<_, Vec<Heading>> = BTreeMap::new();
     for (idx, (head, item)) in toml.as_table_mut().iter_mut().enumerate() {
-        if !matcher.heading.contains(&head.get()) {
+        let mut table_path: Option<Vec<String>> = None;
+        let item_key = head.get();
+        if item_key == TARGET {
+            let mut special_tables = Vec::new();
+            let mut path = vec![item_key];
+            if let Some(table) = item.as_table() {
+                special_tables.push((vec![item_key], table));
+                collect_special_tables(table, &mut path, &mut special_tables, &matcher);
```

Comment:
Perhaps "nested table" would be a better name?

Reply:
changed